test/CheckedC/inferred-bounds/member-base.c

```diff
@@ -145,22 +145,25 @@ void f3(_Array_ptr<struct S> a : bounds(a, a + 7)) {
 // CHECK: `-ImplicitCastExpr {{0x[0-9a-f]+}} '_Array_ptr<struct S>' <LValueToRValue>
 // CHECK: `-DeclRefExpr {{0x[0-9a-f]+}} '_Array_ptr<struct S>' lvalue ParmVar {{0x[0-9a-f]+}} 'a' '_Array_ptr<struct S>'
 
-// CHECK: BinaryOperator {{0x[0-9a-f]+}} '_Ptr<int>' '='
-// CHECK: |-DeclRefExpr {{0x[0-9a-f]+}} '_Ptr<int>' lvalue Var {{0x[0-9a-f]+}} 'p' '_Ptr<int>'
-// CHECK: `-ImplicitCastExpr {{0x[0-9a-f]+}} '_Ptr<int>' <BitCast>
+  *p = 1;
+  p = &(a->f);
+
+// CHECK: BinaryOperator {{0x[0-9a-f]+}} '_Array_ptr<int>' '='
+// CHECK: |-DeclRefExpr {{0x[0-9a-f]+}} '_Array_ptr<int>' lvalue Var {{0x[0-9a-f]+}} 'p' '_Array_ptr<int>'
+// CHECK: `-ImplicitCastExpr {{0x[0-9a-f]+}} '_Array_ptr<int>' <BitCast>
 // CHECK: `-UnaryOperator {{0x[0-9a-f]+}} 'int *' prefix '&'
 // CHECK: `-ParenExpr {{0x[0-9a-f]+}} 'int' lvalue
 // CHECK: `-MemberExpr {{0x[0-9a-f]+}} 'int' lvalue ->f {{0x[0-9a-f]+}}
 // CHECK: `-ImplicitCastExpr {{0x[0-9a-f]+}} '_Array_ptr<struct S>' <LValueToRValue>
 // CHECK:           `-DeclRefExpr {{0x[0-9a-f]+}} '_Array_ptr<struct S>' lvalue ParmVar {{0x[0-9a-f]+}} 'a' '_Array_ptr<struct S>'
 // CHECK: Target Bounds:
 // CHECK: RangeBoundsExpr {{0x[0-9a-f]+}} 'NULL TYPE'
-// CHECK: |-ImplicitCastExpr {{0x[0-9a-f]+}} '_Ptr<int>' <LValueToRValue>
-// CHECK: | `-DeclRefExpr {{0x[0-9a-f]+}} '_Ptr<int>' lvalue Var {{0x[0-9a-f]+}} 'p' '_Ptr<int>'
-// CHECK: `-BinaryOperator {{0x[0-9a-f]+}} '_Ptr<int>' '+'
-// CHECK: |-ImplicitCastExpr {{0x[0-9a-f]+}} '_Ptr<int>' <LValueToRValue>
-// CHECK: | `-DeclRefExpr {{0x[0-9a-f]+}} '_Ptr<int>' lvalue Var {{0x[0-9a-f]+}} 'p' '_Ptr<int>'
-// CHECK: `-IntegerLiteral {{0x[0-9a-f]+}} 'unsigned long long' 1
```

Comment:
This is the most infuriating diff.

`_Array_ptr<T> p : count(i) = expr`, where `i` is a literal, `i` has type `int`
`_Ptr<T> p = expr`, the `1` literal in the range-expanded bounds has type `size_t` i think. We should aim for these to be the same, if we can.